### PR TITLE
Allow Stream<T> and subtypes of it in outputs, refs #76

### PR DESCRIPTION
**Pull request by MichaelRFairhurst**
Use the mostSpecificTypeArgument method API to get the stream type
regardless of the inheritance tree and generics defined within it.

### CODE REVIEW

File: analyzer_plugin/lib/src/tasks.dart [hunk 1]
```diff
@@ -437,9 +441,11 @@ class BuildUnitDirectivesTask extends SourceBasedAnalysisTask
     if (getter != null && getter.type != null) {
       var returntype = getter.type.returnType;
```
Comment (scheglov) [inline analyzer_plugin/lib/src/tasks.dart]: `returnType`?

File: analyzer_plugin/test/tasks_test.dart [hunk 2]
```diff
@@ -620,6 +620,124 @@ class MyComponent {
     errorListener.assertNoErrors();
   }
 
+  void test_outputs_streamIsOk() {
+    String code = r'''
+import '/angular2/angular2.dart';
+import 'dart:async';
+
+@Component(
+    selector: 'my-component',
```
Comment (scheglov) [inline analyzer_plugin/test/tasks_test.dart]: I think this is a compilation error.
Here and below.

File: analyzer_plugin/lib/src/tasks.dart (hunk 1 above)
Comment (MichaelRFairhurst) [inline analyzer_plugin/lib/src/tasks.dart]: Great catch. Looks like that occurred lower in the file too, fixed both

File: analyzer_plugin/test/tasks_test.dart (hunk 2 above)
Comment (MichaelRFairhurst) [inline analyzer_plugin/test/tasks_test.dart]: Wow, lots of these. Was just doing a merge and saw this. Thought I just read the diff wrong! fixed

File: analyzer_plugin/lib/src/tasks.dart [hunk 3]
```diff
@@ -1390,14 +1390,14 @@ class _BuildStandardHtmlComponentsVisitor extends RecursiveAstVisitor {
 
       if (!outputMap.containsKey(name)) {
         if (accessor.isGetter) {
-          var returntype =
+          var returnType =
               accessor.type == null ? null : accessor.type.returnType;
           DartType eventtype = null;
```
Comment (scheglov) [inline analyzer_plugin/lib/src/tasks.dart]: `eventType`

**Pull request merged**